jerry-core/ecma/operations/ecma-container-object.c

```diff
@@ -721,6 +721,31 @@ ecma_op_container_set_weak (ecma_object_t *const key_p, /**< key object */
   ecma_collection_push_back (refs_p, container_value);
 } /* ecma_op_container_set_weak */
 
+/**
+ * Helper method for the Map.prototype.set and Set.prototype.add methods to swap the sign of the given value if needed
+ *
+ * See also:
+ *          ECMA-262 v6, 23.2.3.1 step 6
+ *          ECMA-262 v6, 23.1.3.9 step 6
+ *
+ * @return ecma value
+ */
+static ecma_value_t
+ecma_op_container_set_swap_sign_helper (ecma_value_t this_arg) /*< this arg */
+{
+  if (ecma_is_value_number (this_arg))
+  {
+    ecma_number_t number_value = ecma_get_number_from_value (this_arg);
+
+    if (ecma_number_is_zero (number_value) && ecma_number_is_negative (number_value))
```

Comment:
We should make these an unlikely path as compared to the ecam_number_t range the negative zero is very unlikely to happen, especially multiple times in a map/set